### PR TITLE
clean: optimize post-render code

### PR DESCRIPTION
**Pull request by yxxhero**
clean: optimize post-render code

### CODE REVIEW

File: pkg/state/helmx.go [hunk 1]
```diff
@@ -26,6 +26,23 @@ func (st *HelmState) appendHelmXFlags(flags []string, release *ReleaseSpec) ([]s
 	return flags, nil
 }
 
+// append post-renderer flags to helm flags
+// nolint: unparam
+func (st *HelmState) appendPostRenderFlags(flags []string, release *ReleaseSpec, helm helmexec.Interface) ([]string, error) {
```
Comment (xiaomudk) [inline pkg/state/helmx.go:31]: I think there is no need to return error here

**Review by xiaomudk — APPROVED** [review]

File: pkg/state/helmx.go (hunk 1 above)
Comment (yxxhero) [reply pkg/state/helmx.go:31]: Similar cases will be unified pr for optimization

File: pkg/helmexec/exec.go [hunk 2]
```diff
@@ -258,10 +258,6 @@ func (helm *execer) SyncRelease(context HelmContext, name, chart string, flags .
 		env["HELM_TILLER_HISTORY_MAX"] = strconv.Itoa(context.HistoryMax)
 	}
 
-	if helm.IsHelm3() && helm.postRenderer != "" {
```
Comment (mumoshu) [inline pkg/helmexec/exec.go:261]: Probably we remove the `postRenderer` field out of the helmExec struct now as unused?

File: pkg/state/helmx.go [hunk 3]
```diff
@@ -26,6 +26,23 @@ func (st *HelmState) appendHelmXFlags(flags []string, release *ReleaseSpec) ([]s
...
+	if helm.IsHelm3() {
+		switch {
+		// helm.GetPostRenderer() comes from cmd flag.
+		case helm.GetPostRenderer() != "":
```
Comment (yxxhero) [inline pkg/state/helmx.go:35]: @mumoshu helm.GetPostRenderer() != "" will get postRenderer of HelmExec.

Comment (mumoshu) [reply pkg/state/helmx.go:35]: Ah gotcha! Thanks for clarifying

**Pull request merged**